### PR TITLE
DOC: Update & improve docstrings and type annotations for common methods and classes

### PR DESCRIPTION
**Pull request by BenjaminBossan**
The docstrings of the most user-exposed methods and classes have been updated, or added if not already present. Furthermore, type annotations have been updated or added for those methods and classes.

### CODE REVIEW

File: src/peft/peft_model.py [hunk 1]
```diff
@@ -175,15 +175,16 @@ def save_pretrained(
...
-            selected_adapters (`list(str)`,  *optional*):
+                Whether to save the adapter files in safetensors format, defaults to `True`.
+            selected_adapters (`list` of `str`,  *optional*):
                 A list of adapters to be saved. If `None`, will default to all adapters.
-            save_embedding_layers (`Union[bool, str]`, , *optional*, defaults to `auto`):
+            save_embedding_layers (`Union[bool, str]`, *optional*, defaults to `"auto"`):
                 If `True`, save the embedding layers in addition to adapter weights. If `auto`, checks the common
                 embedding layers `peft.utils.other.EMBEDDING_LAYER_NAMES` in config's `target_modules` when available.
-                Based on it sets the boolean flag. This only works for 🤗 transformers models.
+                Based on it sets, the boolean flag. This only works for 🤗 transformers models.
```
Comment (stevhliu) [inline src/peft/peft_model.py:184]: This was a little difficult to understand. I think it'd be clearer to combine this with the previous sentence about `auto`

```suggestion
                and automatically sets the boolean flag. This only works for 🤗 transformers models.
```

File: src/peft/peft_model.py [hunk 2]
```diff
@@ -544,7 +545,14 @@ def _get_base_model_class(self, is_prompt_tuning=False):
     @contextmanager
     def disable_adapter(self):
         """
-        Disables the adapter module.
+        Context managers that disables the adapter module. Use this to run inference on the base model.
```
Comment (stevhliu) [inline src/peft/peft_model.py:548]:
```suggestion
        Context manager that disables the adapter module. Use this to run inference on the base model.
```

**Review by stevhliu — APPROVED** [review]
Thanks for cleaning these up (most of my suggestions are just minor formatting nits)! ❤️

File: src/peft/tuners/tuners_utils.py [hunk 3]
```diff
@@ -386,11 +399,11 @@ def enable_adapters(self, enabled: bool):
...
             self._disable_adapters = True
 
-    def set_adapter(self, adapter_names: str | list[str]):
-        """Set the active adapter
+    def set_adapter(self, adapter_names: str | list[str]) -> None:
+        """Set the active adapter(s).
 
         Args:
-            adapter_name (str): The name of the adapter to set as active
+            adapter_name (`str` or `list[str]`): Name of the adapter(s) to be activated.
```
Comment (younesbelkada) [inline src/peft/tuners/tuners_utils.py:406]:
```suggestion
            adapter_name (`str` or `List[str]`): Name of the adapter(s) to be activated.
```

**Pull request merged**